### DISCUSSION TITLE
Lesson 13 Member "upgradeTo" not found or not visible after argument-dependent lookup in contract BoxV1

### DISCUSSION THREAD

**Discussion by Dev568145** [Cyfrin/foundry-full-course-cu#1701] [Q&A] [answered]
Hi.

Having trouble figuring out why I'm running into this error while testing:
'''
Error: 
Compiler run failed:
Error (9582): Member "upgradeTo" not found or not visible after argument-dependent lookup in contract BoxV1.
  --> script/UpgradeBox.s.sol:23:9:
   |
23 |         proxy.upgradeTo(address(newBox)); // proxy contract points to this new address
   |      
'''

The two contracts mentioned in the error:

'''
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.20;

import {UUPSUpgradeable} from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
import {Initializable} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";

contract BoxV1 is Initializable, OwnableUpgradeable, UUPSUpgradeable {
    uint256 internal number;

    // we can't use a constructor in a proxy contract

    /// @custom:oz-upgrades-unsafe-allow constructor
    constructor() {
        _disableInitializers(); // disables the initializer check
    }

    function initialize() public initializer {
        __Ownable_init(msg.sender); // sets the owner to msg.sender
        __UUPSUpgradeable_init();
    }

    function setNumber(uint256 _number) external {}

    function getNumber() public view returns (uint256) {
        return number;
    }

    function version() public pure returns (uint256) {
        return 1;
    }

    function _authorizeUpgrade(address newImplementation) internal override {}
}
'''

'''
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.20;

import {Script} from "forge-std/Script.sol";
import {DevOpsTools} from "lib/foundry-devops/src/DevOpsTools.sol";
import {BoxV2} from "../src/BoxV2.sol";
import {BoxV1} from "../src/BoxV1.sol";

contract UpgradeBox is Script {
    function run() external returns (address) {
        address mostRecentlyDeployed = DevOpsTools.get_most_recent_deployment("ERC1967Proxy", block.chainid);

        vm.startBroadcast();
        BoxV2 newBox = new BoxV2(); // new implementation (Logic)
        vm.stopBroadcast();
        address proxy = upgradeBox(mostRecentlyDeployed, address(newBox));
        return proxy;
    }

    function upgradeBox(address proxyAddress, address newBox) public returns (address) {
        vm.startBroadcast();
        BoxV1 proxy = BoxV1(proxyAddress);
        proxy.upgradeTo(address(newBox)); // proxy contract points to this new address
        vm.stopBroadcast();
        return address(proxy);
    }
}
'''
Would be gratefull for any help. Thanks

**Comment by EngrPips** [comment, marked as answer]
Hello @Dev568145, This is addressed in the `chronological update` section of the `Github repo` associated with the course https://github.com/Cyfrin/foundry-full-course-f23/blob/main/chronological-updates.md

**Reply by Dev568145** [reply]
Thanks!

**Reply by EngrPips** [reply]
Anytime, Friend.

**Reply by smilingkylan** [reply]
Unfortunately now that's making it so that `_disableInitializers()` function is an `Undeclared identifier`

**Reply by EngrPips** [reply]
Hey @kylanhurt, You can do without the `_disableInitializers`; everything should still work fine.

**Reply by mustby** [reply]
This thread helped me a lot, I was stuck for WAY too long. Thanks guys

**Reply by EngrPips** [reply]
> This thread helped me a lot, I was stuck for WAY too long. Thanks guys

We are glad this helped you, keep crushing it.
